src/components/base/outline-modal/outline-modal.ts

```diff
@@ -0,0 +1,282 @@
+  area[href]:not([tabindex="-1"]),
+  input:not([disabled]):not([tabindex="-1"]),
+  select:not([disabled]):not([tabindex="-1"]),
+  textarea:not([disabled]):not([tabindex="-1"]),
+  button:not([disabled]):not([tabindex="-1"]),
+  iframe:not([tabindex="-1"]),
+  [tabindex]:not([tabindex="-1"]),
+  [contentEditable=true]:not([tabindex="-1"])
+`;
+
+/**
+ * The Outline Modal component
+ * @element outline-modal
+ * @slot default - The modal contents
+ * @slot outline-modal--trigger - The trigger for the modal
+ * @slot outline-modal--header - The header in the modal
+ * @slot outline-modal--accessibility-description - The accessibility description which is used by screen readers.
+ */
+@customElement('outline-modal')
+export class OutlineModal
+  extends OutlineElement
+  implements OutlineModalInterface
+{
+  static styles: CSSResultGroup = [componentStyles];
+
+  @property({ attribute: false })
+  isOpen = false;
+
+  /**
+   * If we force the user to take an action, the consumer must provide a way to close the modal on their own.
+   */
+  @property({ type: Boolean })
+  shouldForceAction = false;
+
+  @property({ type: String })
+  size?: ModalSize = 'medium';
+
+  render(): TemplateResult {
+    return html`
+      <div
+        id="trigger"
+        tabindex="0"
+        @click="${this.open}"
+        @keydown="${this._handleTriggerKeydown}"
+      >
+        <slot name="outline-modal--trigger"></slot>
+      </div>
+      ${this._overlayTemplate()}
+    `;
+  }
+
+  @state()
+  _hasHeaderSlot: boolean;
+
+  @state()
+  _hasAccessibilityDescriptionSlot: boolean;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._handleSlotChange();
+  }
+
+  private _handleSlotChange(): void {
+    this._hasHeaderSlot =
+      this.querySelector('[slot="outline-modal--header"]') !== null;
+    this._hasAccessibilityDescriptionSlot =
+      this.querySelector(
+        '[slot="outline-modal--accessibility-description"]'
+      ) !== null;
+  }
+
+  private _overlayTemplate(): TemplateResult {
+    let template = html``;
+
+    if (this.isOpen) {
+      template = html`
+        <div
+          id="overlay"
+          tabindex="-1"
+          class="${this.size}"
+          @click="${this._handleOverlayClick}"
+          @keydown="${this._handleOverlayKeydown}"
+        >
+          <div
+            id="container"
+            role="dialog"
+            aria-modal="true"
+            aria-labelledby="${ifDefined(
+              this._hasHeaderSlot ? 'header' : undefined
+            )}"
+            aria-describedby="${ifDefined(
+              this._hasAccessibilityDescriptionSlot
+                ? 'accessibility-description'
+                : undefined
+            )}"
+          >
+            <div id="header">
+              <slot
+                id="title"
+                name="outline-modal--header"
+                @slotchange="${this._handleSlotChange}"
+              ></slot>
+              ${this.shouldForceAction
```

Comment:
Can we place `this.shouldForceAction` in it's own template function? See [here](https://github.com/phase2/outline/pull/60#discussion_r679571323) for examples